core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -247,6 +249,15 @@ private static ServiceRequestContext serviceRequestContext() {
         return current != null ? current.root() : null;
     }
 
+    @Nullable
+    private static AttributesGetters getAttributes(@Nullable ServiceRequestContext ctx) {
```

Comment:
nit: could we move this method down after the below constructor that calls this method for readability?